src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+                    throw anException;
+                }
+            }
+        };
+        this.settingNameTooltipTextMap.put(this.chainFragmentLengthSetting.getName(),
+                Message.get("AlkylStructureFragmenter.chainFragmentLengthSetting.tooltip"));
+        this.dissectSpiroCarbonSetting = new SimpleBooleanProperty(this, "Dissect spiro carbon setting",
+                AlkylStructureFragmenter.DISSECT_SPIRO_CARBON_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.dissectSpiroCarbonSetting.getName(),
+                Message.get("AlkylStructureFragmenter.dissectSpiroCarbonSetting.tooltip"));
+        this.dissectRingsSetting = new SimpleBooleanProperty(this, "Dissect rings setting",
+                AlkylStructureFragmenter.DISSECT_RINGS_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.dissectRingsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.dissectRingsSetting.tooltip"));
+        this.preserveRingMaxSizeSetting = new SimpleEnumConstantNameProperty(this, "Preserve ring max size setting (preserves if Dissect rings setting is set false)",
+                AlkylStructureFragmenter.Preserve_Ring_MAX_SIZE_OPTION_DEFAULT.name(),
+                PreserveRingMaxSizeOption.class) {
+            @Override
+            public void set(String newValue) throws NullPointerException, IllegalArgumentException {
+                try {
+                    super.set(newValue);
+                } catch (NullPointerException | IllegalArgumentException anException) {
+                    AlkylStructureFragmenter.this.logger.log(Level.WARNING, anException.toString(), anException);
+                    GuiUtil.guiExceptionAlert("Illegal Argument", "Illegal Argument was set", anException.toString(), anException);
+                    //re-throws the exception to properly reset the binding
+                    throw anException;
+                }
+            }
+        };
+        this.settingNameTooltipTextMap.put(this.preserveRingMaxSizeSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingMaxSizeSetting.tooltip"));
+        this.preserveRingSystemSetting = new SimpleBooleanProperty(this, "Preserve ring system setting (preserves if Dissect rings setting is set false)",
+                AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.preserveRingSystemSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingSystemSetting.tooltip"));
+        this.preserveRingSystemMaxSetting = new SimpleIntegerProperty(this, "Preserve ring system up to maximum size",
+                AlkylStructureFragmenter.PRESERVE_RING_SYSTEM_MAX_OPTION_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.preserveRingSystemMaxSetting.getName(),
+                Message.get("AlkylStructureFragmenter.preserveRingSystemMaxSetting.tooltip"));
```

Comment:
About the settings: I guess most of the settings are Dummies right now (see my general comment about that), but when they gain function, you have to think about which combinations are allowed and which aren't. Because I would guess that some of your ring-related settings don't make sense to be mixed in certain ways.